### PR TITLE
sql: add SHOW EXPERIMENTAL_FINGERPRINTS FROM VIRTUAL CLUSTER

### PR DESCRIPTION
**Pull request by stevendanna**
This adds a user-facing interface for tenant fingerprinting that does not require crdb_internal functions.

We've moved the underlying implementation to a function that lives on the planner so that it can be shared from both call sites.

Additionally,

- we now issue the ExportRequests in parallel, and
- we allow users with MANAGETENANT to issue the fingerprint command.

Epic: none

Release note: None

### CODE REVIEW

File: pkg/sql/catalog/colinfo/result_columns.go [hunk 1]
```diff
@@ -192,12 +192,20 @@ var ShowSyntaxColumns = ResultColumns{
...
 // ShowFingerprintsColumns are the result columns of a
-// SHOW EXPERIMENTAL_FINGERPRINTS statement.
+// SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE statement.
 var ShowFingerprintsColumns = ResultColumns{
 	{Name: "index_name", Typ: types.String},
 	{Name: "fingerprint", Typ: types.String},
 }
 
+// ShowFingerprintsColumns are the result columns of a
+// SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE statement.
```
Comment (msbutler) [inline pkg/sql/catalog/colinfo/result_columns.go:202]: nit:  s/TABLE/TENANT/r

**Review by msbutler — APPROVED** [review]
Looks great! If you're feeling ambitious, you could update the c2c roachtests to use the syntax. But also it doesn't really matter bc crdb_internal.fingerprint is the same.

**Review by knz — APPROVED** [review]
> Reviewed 5 of 21 files at r1, 4 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111190)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @stevendanna)

**Comment by stevendanna** [comment]
bors r=knz,msbutler

**Pull request merged**